pkg/server/webhook.go

```diff
@@ -348,21 +353,25 @@ func setServiceAccount(initContainers []corev1.Container, containers []corev1.Co
 // this does _not_ return patches; this is intended to be used only on containers defined
 // in the injection config, so the resources do not exist yet in the k8s api (thus no patch needed)
 func mergeEnvVars(envs []corev1.EnvVar, containers []corev1.Container) []corev1.Container {
+	hasEnvVar := func(existing []corev1.EnvVar, add corev1.EnvVar) bool {
+		for _, v := range existing {
+			// if any of the existing volumes have the same name as test.Name, skip
+			// injecting it
+			if v.Name == add.Name {
+				return true
+			}
+		}
+		return false
+	}
 	mutatedContainers := []corev1.Container{}
 	for _, c := range containers {
 		for _, newEnv := range envs {
 			// check each container for each env var by name.
 			// if the container has a matching name, dont override!
-			skip := false
```

Comment:
this logic was confusing and the control flow also was suspect, so i moved it into a helper func